packages/server/src/base/schema.ts

```diff
@@ -0,0 +1,15 @@
+import Joi from 'joi'
+
+interface SchemaObject {
+  [key: string]: Joi.AnySchema
+}
+
+export const ReqSchema = (
+  schema?: { body?: SchemaObject; params?: SchemaObject; query?: SchemaObject } | undefined
+) => {
+  return Joi.object({
+    body: Joi.object(schema?.body || {}).required(),
+    params: Joi.object(schema?.params || {}).required(),
+    query: Joi.object(schema?.query || {}).required()
+  }).required()
+}
```

Comment:
This is the new function to define request schemas. It always validate the body, params, and query parameters.